### DISCUSSION TITLE
DevServices : quarkus.datasource.devservices.container-properties

### DISCUSSION THREAD

**Discussion by adarshahd** [quarkusio/quarkus#24956] [Q&A]
I would like to know what are the available properties for the container-properties.

I have a migration sql inside resources/db/migration, which specifies the database name (ex: CREATE SCHEMA IF NOT EXISTS `posts`). It results in error during migration because Flyway assumes database name to be 'default' (From DevServices).

I guessed several different options for the container-properties so that I could change the database name, but doesn't work. I would like to know the options. Thanks in advance.

As a workaround if I remove the database name in the migration sql, then everything works fine.

[a comment by an automated account is not shown]

**Comment by pcaceresf** [comment]
Devservices is a really problem, by default always FAILS, i spendt a lot of time with this fucking problem.

**Reply by gsmet** [reply]
@pcaceresf please be polite. Also did you create GitHub issues explaining your problems? It works for a lot of people.

**Reply by adarshahd** [reply]
@pcaceresf hey, calm down! not  sure what exactly is your problem, but I feel generally Quarkus is really good framework. You could specify your issues and community can take a look.

**Comment by gsmet** [comment]
@adarshahd could you create an issue with a small Maven reproducer explaining exactly what your problem is? We will have a look.

**Reply by adarshahd** [reply]
Hello, First of all, thank for great work on Quarkus. may be I should have been more specific with the issue. I would like to configure database name, user and password for the database used for the testcontainers. Something like below with container-properties

```
%dev.quarkus.datasource.devservices.image-name=mariadb:10.7
%dev.quarkus.datasource.devservices.port=33306
%dev.quarkus.datasource.devservices.container-properties.MARIADB_DATABASE=users
%dev.quarkus.datasource.devservices.container-properties.MARIADB_USER=root
%dev.quarkus.datasource.devservices.container-properties.MARIADB_PASSWORD=password
```
But these are not explained enough in the document, and do not work with the above parameters

If you still need a reproducer let me know, I can send one

**Reply by gsmet** [reply]
So for user and password, you can use the regular datasource config for user and password and they will be taken into account.

For the database name, it's a bit more cumbersome: the database name is forced to the name of the datasource so you can tweak it but that forces you to use a named datasource with this name.

Maybe we should provide a way to override it. There is an ongoing PR in this area, I will see if I can improve the situation in this area.

**Reply by adarshahd** [reply]
Thanks. Good to hear that. However your suggestions do not answer issues completely. I would like little more documentation on how to use the container-properties parameters in the application.properties file. It could also be taken as a doc issue.

**Reply by gsmet** [reply]
Maybe it wasn't clear but I totally agree on that. I added something on my TODO list.

**Reply by adarshahd** [reply]
Thank you very much!

**Reply by ghost** [reply]
@gsmet I'm not sure if the issue described by @adarshahd is the same one I experienced earlier today, but it sure seems like it.  Earlier today, I discovered that DevServices will attempt to use the name of the declared datasource as the default schema when performing queries.  For example, a MySQL datasource declared like this:
```
quarkus.datasource.test.reactive=false
quarkus.datasource.test.db-kind=mysql
quarkus.datasource.test.username=root
quarkus.datasource.test.password=taco
```

And injected into code like this:
```
@ApplicationScoped
class PotatoTypeReadOnlyRepository : IPotatoTypeReadRepository {
	// Dependencies
	private val testDataSource: AgroalDataSourcel

	// Constructors
	@Inject
	constructor(@DataSource("test") testDataSource: AgroalDataSource) {
		this.testDataSource = testDataSource
	}
}
```

Will attempt to use `test` as the database schema.  Which means when I submit a query like this:
```
val sqlQuery = "select * from PotatoTypes where NAME = ? order by NAME limit 2"
```

It attempts to query the `test.PotatoTypes` table.  And if that table exists on the `POTATO_DB` schema, and not a schema named `test`, the query fails.

I was kinda expecting there to be a configuration class property that let me set the name of the schema that's associated with a datasource.  Something like this:
```
quarkus.datasource.test.schema=POTATO_DB
```

If it's worth writing up the behavior I experienced as a separate discussion, say the word & I'll get that done on Tuesday.  I already have most of the sample repository code ready -- I just need to make a few changes & then get that code up into a new GitHub repo.

**Reply by gsmet** [reply]
@adarshahd what I did in the last commit of this PR should make it easier for you: https://github.com/quarkusio/quarkus/pull/25288 . It's going to be automatically documented in the configuration reference.

**Reply by adarshahd** [reply]
@gsmet I can see these changes will make it easy to modify username/password/schema names. Thank you very much again!
